File: Basic/bjgood01/Queue/10845.py
import sys
# 큐를 구현할때는 deque 라이브러리를 사용한다
from collections import deque
n = int(input())
queue = deque()
for i in range(n):
    input = sys.stdin.readline().strip()
    a = input.split()  # 띄어쓰기를 기준으로 나눈다
    if a[0] == 'push':
        queue.append(int(a[1]))
    elif a[0] == 'pop':
        if len(queue) == 0:
            print(-1)
        else:
            print(queue.popleft())
    elif a[0] == 'size':
        print(len(queue))
    elif a[0] == 'empty':
        if len(queue) == 0:
            print(1)
        else:
            print(0)
    elif a[0] == 'front':
        if len(queue) == 0:
            print(-1)
        else:
            print(queue[0])
    elif a[0] == 'back':
        if len(queue) == 0:
            print(-1)
        else:
            print(queue[-1])

File: 2023-2/Basic/gyeongtae/Queue/10845.cpp
#include <iostream>
#include <queue>

using namespace std;


int main() {
    int a,b,c;
    string n, m;
    queue <int> q;
    cin >> a;
    for (int i = 0;i < a;i++) {
        cin >> n;
        if (n == "push") {
            cin >> b;
            q.push(b);
        }
        else if (n == "pop") {
            if (!q.empty()) {
                cout << q.front() << endl;
                q.pop();
            }
            else {
                cout << "-1" << endl;
            }
        }
        else if (n == "size") {
            cout << q.size() << endl;
        }
        else if (n == "empty") {
            if (q.empty()) {
                cout << "1" << endl;
            }
            else {
                cout << "0" << endl;
            }
        }
        else if (n == "front") {
            if (!q.empty()) {
                cout << q.front() << endl;
            }
            else {
                cout << "-1" << endl;
            }
        }
        else if (n == "back") {
            if (!q.empty()) {
                cout << q.back() << endl;
            }
            else {
                cout << "-1" << endl;
            }
        }
    }

}
